contracts/vault/src/error.rs

```diff
@@ -0,0 +1,33 @@
+use cosmwasm_std::{StdError, Uint128};
+use cw_utils::{ParseReplyError, PaymentError};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ContractError {
+    #[error("{0}")]
+    Std(#[from] StdError),
+
+    #[error("{0}")]
+    Payment(#[from] PaymentError),
+
+    #[error("{0}")]
+    ParseReply(#[from] ParseReplyError),
+
+    #[error("Unauthorized")]
+    Unauthorized {},
+
+    #[error("Claim is locked, only {0} can be unbonded")]
+    ClaimsLocked(Uint128),
+
+    #[error("The address doesn't have sufficient balance for this operation")]
+    InsufficentBalance,
+
+    #[error("The leinholder doesn't have any claims")]
+    UnknownLeinholder,
```

Comment:
Here and elsewhere, do you mean 'lien'?

```suggestion
    #[error("The lien holder doesn't have any claims")]
    UnknownLienholder,
```

Reply:
Seems `leinholder` is an actual word from a quick search...

Reply:
Hmm, it seems to me that the right or more common expression is '[lien](https://www.thefreedictionary.com/Lein) holder'.

Reply:
Mauro is correct. Lien is what I intended:

"a right to keep possession of property belonging to another person until a debt owed by that person is discharged"